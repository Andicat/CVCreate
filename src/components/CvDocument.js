import React from 'react';
import PropTypes from 'prop-types';
import {Transition} from "react-transition-group";
import {connect} from 'react-redux';
import CvBlock from './CvBlock';
import Transform from './Transform';
import {cvBlock_activate} from '../redux/cvDataAC';
import {createStyle} from '../modules/utils';

//документ
class CvDocument extends React.PureComponent {

    static propTypes = {
        blocks: PropTypes.array,
        activeBlock: PropTypes.object,
        activeBlocksId: PropTypes.array,
        showPanel: PropTypes.bool,
        stylePage: PropTypes.object,
    };

    //активируем документ
    onClick = (evt) => {
        if (evt.target === evt.currentTarget) {
            this.props.dispatch(cvBlock_activate(null,null));
        }
    }

    render () {
        let cvBlocksCode = this.props.blocks.map( b => {
            let activeIndex = this.props.activeBlocksId.findIndex(ab => ab===b.id);
            return <CvBlock key={b.id} id={b.id} data={b} activeIndex={activeIndex} activeElementId={activeIndex>=0?this.props.activeElementId:null} editable={true}></CvBlock>
        });
        
        return (
            <div className='cv-container' onClick={this.onClick}>
                {(this.props.activeBlock && this.cv) && (
                    <Transition in={this.props.showPanel} timeout={{ enter: 1000, exit: 1000 }}>
                        {stateName => {
                            return <Transform block={this.props.activeBlock} cv={this.cv} transitionClass={stateName}/>
                        }}
                    </Transition>
                )}
                <div className='cv' style={createStyle(this.props.stylePage)} onClick={this.onClick} ref={(f) => this.cv = f}>
                    {cvBlocksCode}
                </div>
            </div>
        );
    }
}

const mapStateToProps = function (state) {
    return {
        blocks: state.cvData.blocks,
        activeBlocksId: state.cvData.activeBlocksId,
        activeElementId: state.cvData.activeElementId
    };
};
  
export default connect(mapStateToProps)(CvDocument);
