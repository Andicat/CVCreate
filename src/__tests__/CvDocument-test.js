

import {renderComponentWithStore, storeEmpty, storeWithData} from './utils-test';

import CvDocument from '../components/CvDocument';

let component;

describe('CvDocument render', () => {

    it('Рендер без данных', () => {
        component = renderComponentWithStore(CvDocument,storeEmpty,);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер с данными', () => {
        component = renderComponentWithStore(CvDocument,storeWithData);
        expect(component.toJSON()).toMatchSnapshot();
    });

    it('Рендер с активным блоком', () => {
        let props = {activeBlock: {'style': {'s00_font': 'Roboto','s08_padding': {'bottom': 0,'left': 1,'right': 0,'top': 0},'s03_bold': true,'s04_italic': false,'s05_center': false,'s01_color': '#1bac38','s06_uppercase': false,'s07_underline': false,'s02_fontsize': '40'},'type': 'text','text': 'Big text','width': 149,'height': 66,'positionTop': 52.08598862935976,'positionLeft': 290.08598862935975,'id': 2}};
        component = renderComponentWithStore(CvDocument,storeWithData,props);
        let blockCv = component.root.find( el => el.props.className==='cv');
        const evt = {'currentTarget': blockCv};
        blockCv.props.onClick(evt);
        expect(component.toJSON()).toMatchSnapshot();
    });
});
