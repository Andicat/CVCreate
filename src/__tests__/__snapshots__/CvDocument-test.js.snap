// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CvDocument render Рендер без данных 1`] = `
<div
  className="cv-container"
>
  <div
    className="cv"
    onClick={[Function]}
    style={Object {}}
  />
</div>
`;

exports[`CvDocument render Рендер с активным блоком 1`] = `
<div
  className="cv-container"
>
  <div
    className="cv"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "66px",
          "left": "290.08598862935975px",
          "top": "52.08598862935976px",
          "width": "149px",
        }
      }
    >
      <span
        className="cv__element cv__element--text"
        contentEditable={true}
        data-elem={true}
        onBlur={[Function]}
        onClick={[Function]}
        style={
          Object {
            "color": "#1bac38",
            "fontFamily": "Roboto",
            "fontSize": "40px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "padding": "0px 0px 0px 1px",
            "textAlign": "start",
            "textDecoration": "none",
            "textTransform": "none",
          }
        }
        suppressContentEditableWarning={true}
      >
        Big text
      </span>
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "142px",
          "left": "60.43650934333691px",
          "top": "44.43650934333691px",
          "width": "182px",
        }
      }
    >
      <img
        alt=""
        className="cv__element cv__element--image"
        data-elem={true}
        onClick={[Function]}
        style={
          Object {
            "file": "",
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="cv__block cv__block--active cv__block--active-first entered"
      onClick={[Function]}
      style={
        Object {
          "height": "62px",
          "left": "303.173880178947px",
          "top": "142.173880178947px",
          "width": "119px",
        }
      }
    >
      <div
        className="cv__element cv__element--group"
        style={Object {}}
      >
        <span
          className="cv__element cv__element--text cv__element--active"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "20px",
              "fontStyle": "normal",
              "fontWeight": "bold",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          Your header
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "16px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          your text
        </span>
      </div>
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "50px",
          "left": "72.88890908532193px",
          "top": "248.88890908532193px",
          "width": "179px",
        }
      }
    >
      <div
        className="cv__element cv__element--dots-row"
        data-elem={true}
        onClick={[Function]}
        style={
          Object {
            "addcolor": "#E6E6E6",
            "addcount": 3,
            "maincolor": "#E05B49",
            "maincount": 5,
            "radius": 10,
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E6E6E6",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E6E6E6",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E6E6E6",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
      </div>
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "161px",
          "left": "322.66776857728195px",
          "top": "238.66776857728195px",
          "width": "171px",
        }
      }
    >
      <div
        className="cv__element cv__element--figure"
        data-elem={true}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#6AABB5",
            "borderRadius": "50%",
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "111px",
          "left": "487.9319841589094px",
          "top": "92.93198415890942px",
          "width": "252px",
        }
      }
    >
      <div
        className="cv__element cv__element--group"
        style={Object {}}
      >
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "18px",
              "fontStyle": "normal",
              "fontWeight": "bold",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          Your position
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "18px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          Company
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "16px",
              "fontStyle": "italic",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          period
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "16px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          your competencies and results
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`CvDocument render Рендер с данными 1`] = `
<div
  className="cv-container"
>
  <div
    className="cv"
    onClick={[Function]}
    style={Object {}}
  >
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "66px",
          "left": "290.08598862935975px",
          "top": "52.08598862935976px",
          "width": "149px",
        }
      }
    >
      <span
        className="cv__element cv__element--text"
        contentEditable={true}
        data-elem={true}
        onBlur={[Function]}
        onClick={[Function]}
        style={
          Object {
            "color": "#1bac38",
            "fontFamily": "Roboto",
            "fontSize": "40px",
            "fontStyle": "normal",
            "fontWeight": "bold",
            "padding": "0px 0px 0px 1px",
            "textAlign": "start",
            "textDecoration": "none",
            "textTransform": "none",
          }
        }
        suppressContentEditableWarning={true}
      >
        Big text
      </span>
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "142px",
          "left": "60.43650934333691px",
          "top": "44.43650934333691px",
          "width": "182px",
        }
      }
    >
      <img
        alt=""
        className="cv__element cv__element--image"
        data-elem={true}
        onClick={[Function]}
        style={
          Object {
            "file": "",
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="cv__block cv__block--active cv__block--active-first entered"
      onClick={[Function]}
      style={
        Object {
          "height": "62px",
          "left": "303.173880178947px",
          "top": "142.173880178947px",
          "width": "119px",
        }
      }
    >
      <div
        className="cv__element cv__element--group"
        style={Object {}}
      >
        <span
          className="cv__element cv__element--text cv__element--active"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "20px",
              "fontStyle": "normal",
              "fontWeight": "bold",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          Your header
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "16px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          your text
        </span>
      </div>
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "50px",
          "left": "72.88890908532193px",
          "top": "248.88890908532193px",
          "width": "179px",
        }
      }
    >
      <div
        className="cv__element cv__element--dots-row"
        data-elem={true}
        onClick={[Function]}
        style={
          Object {
            "addcolor": "#E6E6E6",
            "addcount": 3,
            "maincolor": "#E05B49",
            "maincount": 5,
            "radius": 10,
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E05B49",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E6E6E6",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E6E6E6",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
        <div
          style={
            Object {
              "backgroundColor": "#E6E6E6",
              "borderRadius": "50%",
              "height": "10px",
              "margin": "0 5px",
              "width": "10px",
            }
          }
        />
      </div>
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "161px",
          "left": "322.66776857728195px",
          "top": "238.66776857728195px",
          "width": "171px",
        }
      }
    >
      <div
        className="cv__element cv__element--figure"
        data-elem={true}
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#6AABB5",
            "borderRadius": "50%",
            "opacity": 1,
          }
        }
      />
    </div>
    <div
      className="cv__block entered"
      onClick={[Function]}
      style={
        Object {
          "height": "111px",
          "left": "487.9319841589094px",
          "top": "92.93198415890942px",
          "width": "252px",
        }
      }
    >
      <div
        className="cv__element cv__element--group"
        style={Object {}}
      >
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "18px",
              "fontStyle": "normal",
              "fontWeight": "bold",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          Your position
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "18px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          Company
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "16px",
              "fontStyle": "italic",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          period
        </span>
        <span
          className="cv__element cv__element--text"
          contentEditable={true}
          data-elem={true}
          onBlur={[Function]}
          onClick={[Function]}
          style={
            Object {
              "color": "#000000",
              "fontFamily": "Roboto",
              "fontSize": "16px",
              "fontStyle": "normal",
              "fontWeight": "normal",
              "padding": "0px 0px 0px 1px",
              "textAlign": "start",
              "textDecoration": "none",
              "textTransform": "none",
            }
          }
          suppressContentEditableWarning={true}
        >
          your competencies and results
        </span>
      </div>
    </div>
  </div>
</div>
`;
